### DISCUSSION TITLE
ActionView::Template::Error File to import not found or unreadable: bootstrap-email

### DISCUSSION THREAD

**Discussion by anikettstiwari** [bootstrap-email/bootstrap-email#143] [Help]
After upgrading  bootstrap-email from (0.2.5) to (1.0.2) it gives the following error. 

Rails 6.1
Ruby 3.0
bootstrap-email (1.0.2)

my index.scss

/* ...
*= require_self
*= require_tree ..
*/

@import 'bootstrap-email';

Failure/Error: @import 'bootstrap-email';

     ActionView::Template::Error:
       Error: File to import not found or unreadable: bootstrap-email.
               on line 6:1 of app/assets/stylesheets/mailer/index.scss
       >> @import 'bootstrap-email';

          ^
     # ./app/assets/stylesheets/mailer/index.scss:6
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sassc-2.4.0/lib/sassc/engine.rb:50:in `render'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sassc-rails-2.1.2/lib/sassc/rails/template.rb:40:in `block in call'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/utils.rb:138:in `module_include'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sassc-rails-2.1.2/lib/sassc/rails/template.rb:39:in `call'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:84:in `call_processor'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:66:in `block in call_processors'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:65:in `reverse_each'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:65:in `call_processors'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:22:in `block in <class:CompositeProcessor>'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:33:in `call'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:84:in `call_processor'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:66:in `block in call_processors'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:65:in `reverse_each'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:65:in `call_processors'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:182:in `load_from_unloaded'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:59:in `block in load'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:337:in `fetch_asset_from_dependency_cache'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:43:in `load'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/cached_environment.rb:44:in `load'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/bundle.rb:27:in `call'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:84:in `call_processor'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:66:in `block in call_processors'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:65:in `reverse_each'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/processor_utils.rb:65:in `call_processors'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:182:in `load_from_unloaded'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:59:in `block in load'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:337:in `fetch_asset_from_dependency_cache'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/loader.rb:43:in `load'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/cached_environment.rb:44:in `load'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/base.rb:81:in `find_asset'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-4.0.2/lib/sprockets/base.rb:119:in `[]'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:358:in `find_asset'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:337:in `digest_path'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:326:in `asset_path'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:103:in `block in resolve_asset_path'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:243:in `block in resolve_asset'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:242:in `each'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:242:in `detect'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:242:in `resolve_asset'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:102:in `resolve_asset_path'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:80:in `compute_asset_path'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:186:in `block in stylesheet_link_tag'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:184:in `map'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/sprockets-rails-3.2.2/lib/sprockets/rails/helper.rb:184:in `stylesheet_link_tag'
     # ./app/views/job_mailer/job_published_notification.html.erb:1:in `_app_views_job_mailer_job_published_notification_html_erb__1991425911532098642_68680'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/bootstrap-email-1.0.2/lib/bootstrap-email/rails/action_mailer.rb:5:in `bootstrap_mail'
     # ./app/mailers/job_mailer.rb:91:in `validate_and_notify'
     # ./app/mailers/job_mailer.rb:12:in `job_published_notification'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/actiontext-6.1.4.1/lib/action_text/rendering.rb:20:in `with_renderer'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/actiontext-6.1.4.1/lib/action_text/engine.rb:59:in `block (4 levels) in <class:Engine>'
     # <internal:kernel>:90:in `tap'
     # ./spec/mailers/job_mailer_spec.rb:18:in `block (3 levels) in <top (required)>'
     # ./spec/rails_helper.rb:78:in `block (3 levels) in <top (required)>'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/database_cleaner-core-2.0.1/lib/database_cleaner/strategy.rb:30:in `cleaning'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/database_cleaner-core-2.0.1/lib/database_cleaner/cleaners.rb:34:in `block (2 levels) in cleaning'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/database_cleaner-core-2.0.1/lib/database_cleaner/cleaners.rb:35:in `cleaning'
     # ./spec/rails_helper.rb:77:in `block (2 levels) in <top (required)>'
     # /Users/aniket_tiwari/.rvm/gems/ruby-3.0.0@project/gems/webmock-3.14.0/lib/webmock/rspec.rb:37:in `block (2 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # SassC::SyntaxError:
     #   Error: File to import not found or unreadable: bootstrap-email.
     #           on line 6:1 of app/assets/stylesheets/mailer/index.scss
     #   >> @import 'bootstrap-email';
     #
     #      ^
     #   ./app/assets/stylesheets/mailer/index.scss:6

@stuyam Can you please look into the issue?

**Comment by stuyam** [comment]
I would suggest checking the usage guide for Rails for Bootstrap email v1+ as the setup has changed: https://bootstrapemail.com/docs/usage#ruby-on-rails

I think the main change for you is that you no longer need to link the bootstrap email file from the base HTML layout so verify that change.

If you have custom CSS or want to customize checkout the configuration options: https://bootstrapemail.com/docs/configure

I should have better upgrade guide for the Rails portion 😓 

PS: a lot of class and such have changed so also check out: https://bootstrapemail.com/docs/whats-new for changes for that.

**Comment by hananamar** [comment]
I was also struggling with this error message, and I think it's due to the same misunderstanding of the docs here: https://bootstrapemail.com/docs/configure

I would add a link tag to the top mailer layout with my custom CSS, and I thought that file needed to have the line `//= @import bootstrap-email;`, but that's not the case.

What actually I needed to do was create a file called `bootstrap-email.scss` in the stylesheets folder, with the same custom CSS, and not use a link tag at all.
